percy-csharp-selenium/Percy.cs

```diff
@@ -15,63 +15,82 @@ namespace percy_csharp_selenium
    */
     public class Percy
     {
-
-        // We'll expect this file to exist at the root of our classpath, as a resource.
-        private static readonly String AGENTJS_FILE = "percy-agent.js";
-
         // Selenium WebDriver we'll use for accessing the web pages to snapshot.
         private IWebDriver driver;
 
-        // The JavaScript contained in percy-agent.js
-        private String percyAgentJs;
+        // The JavaScript contained in dom.js
+        private String domJs = "";
+
+        // Maybe get the CLI server address if not Set the CLI server address 
+        //could be moved to percy-csharp-selenium Environment
+        private String PERCY_SERVER_ADDRESS = System.Environment.GetEnvironmentVariable("PERCY_SERVER_ADDRESS")!=null?
+                System.Environment.GetEnvironmentVariable("PERCY_SERVER_ADDRESS") : "http://localhost:5338";
 
         // Environment information like the programming language, browser, & SDK versions
         private Environment env;
 
-        // Is the Percy Agent process running or not
-        private bool percyIsRunning = true;
+        // Is the Percy server running or not
+        private Boolean isPercyEnabled;
+
+        // HttpClient is intended to be instantiated once per application, rather than per-use.
+        private static readonly HttpClient client = new HttpClient();
 
         /**
              * @param driver The Selenium WebDriver object that will hold the browser
              *               session to snapshot.
         */
         public Percy(IWebDriver driver)
         {
-            this.driver = driver;
+            this.driver = driver; 
             this.env = new Environment(driver);
-            this.percyAgentJs = LoadPercyAgentJsAsync().Result;
+            isPercyEnabled = Healthcheck().Result;
         }
 
         /**
-             * Attempts to load percy-agent.js from `http://localhost:5338/percy-agent.js`.
-             *
-             * This JavaScript is critical for capturing snapshots. It serializes and captures
-             * the DOM. Without it, snapshots cannot be captured.
+        * Checks to make sure the local Percy server is running. If not, disable Percy.
         */
-        private async System.Threading.Tasks.Task<string> LoadPercyAgentJsAsync()
-        {
-            try
-            {
-                //Creating a HttpGet object
-                using HttpClient client = new HttpClient();
-                var response = await client.GetAsync("http://localhost:5338/" + AGENTJS_FILE);
-
+        private async Task<Boolean> Healthcheck() {
```

Comment:
can we return simple boolean `true` or `false` from this method
and lets do `Healthcheck().Result;` currently done in constructor inside this method itself.